src/redisraft.c

```diff
@@ -604,6 +653,8 @@ static void handleRedisCommand(RedisRaftCtx *rr,
         if (rr->config->quorum_reads) {
             RaftReq *req = RaftReqInit(ctx, RR_REDISCOMMAND);
             RaftRedisCommandArrayMove(&req->r.redis.cmds, cmds);
+            /* need to save asking flag when we move the cmds */
+            req->r.redis.cmds.asking = cmds->asking;
```

Comment:
Maybe this should be part of RaftRedisCommandArrayMove(). Is there an exception to this? Whenever we move a command into the array, we also adopt asking state?

Reply:
possibly.  we dont want to overwrite state each time from true to false (I think) thought that could be solved via a cmds.asking |= cmds->asking within it.

Reply:
> we dont want to overwrite state each time from true to false

Not sure I understand. What is the problem here? Also, we are going to set asking true once because if client is in asking mode, it will stay in asking mode forever, right?  So, once we are in asking mode, we'll never get "asking=false", no? (We may need to validate this somewhere though).